### PR TITLE
3.4 Language evolution

### LINKED ISSUE #5811: [3.4] Add Dart language evolution entry
Add 3.4 section to [Dart language evolution](https://dart.dev/guides/language/evolution) page.

There seem to be 3 language changes in the [changelog](https://github.com/dart-lang/sdk/blob/main/CHANGELOG.md#340).

### PR DESCRIPTION
**Pull request by MaryaBelanger**
Fixes #5811

### CODE REVIEW

File: src/content/guides/language/evolution.md [hunk 1]
```diff
@@ -41,6 +41,18 @@ on the Dart language GitHub repo.
...
+### Dart 3.4
+_Released 14 May 2024_
+| [Dart 3.4 announcement](https://medium.com/dartlang/dart-3-4-bd8d23b4462a)
+
+Dart 3.4 made several improvements related to type analysis. These include:
+
+* Improvements to the type analysis of conditional expressions,
+  if-null expressions and assignments, and switch expressions.
+* Aligning the pattern context type schema for cast patterns with the spec.
+* Making the type schema for the null-aware spread operator nullable for maps
```
Comment (parlough) [inline src/content/guides/language/evolution.md:53]:
```suggestion
* Making the type schema for the null-aware spread operator (`...?`) nullable for maps
```

File: src/content/guides/language/evolution.md [hunk 2]
```diff
@@ -41,6 +41,18 @@ on the Dart language GitHub repo.
...
+  and set literals, to match the behavior for list literals.
```
Comment (parlough) [inline src/content/guides/language/evolution.md:54]:
```suggestion
  and set literals, to match the behavior of list literals.
```

**Review by parlough — APPROVED** [review]
Thanks!!

**Review by stereotype441 — APPROVED** [review]
+1 to @parlough's suggestions

**Pull request merged**